tests/Plugin_Context_Tests.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Tests for the Plugin_Context class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Plugin_Context;
+
+class Plugin_Context_Tests extends WP_UnitTestCase {
+	public function set_up() {
+		parent::set_up();
+
+		$this->plugin_name    = basename( dirname( __DIR__ ) );
+		$this->plugin_context = new Plugin_Context( $this->plugin_name . '/plugin-check.php' );
+	}
+
+	public function test_basename() {
+		$this->assertSame( $this->plugin_name . '/plugin-check.php', $this->plugin_context->basename() );
+	}
+
+	public function test_path() {
+		$this->assertSame( $this->plugin_name . '/', $this->plugin_context->path() );
+	}
+
+	public function test_path_with_parameter() {
+		$this->assertSame( $this->plugin_name . '/another/folder', $this->plugin_context->path( '/another/folder' ) );
+	}
```

Comment:
@jjgrainger Just spotting this post-merge: This still doesn't fully address my original feedback: I would think we need to use `WP_PLUGIN_DIR . '/' . $this->plugin_name` instead of just `$this->plugin_name`. Have you tried that, were there any problems? At least in theory this still looks unstable to me without that bit.

Let me know what you think. If my concerns are valid, we can fix this throughout in a quick follow up PR.

cc @mukeshpanchal27

Reply:
The `plugin_dir_path( $this->main_file )` return plugin directory slug instead of the full relative path.

Reply:
Thanks both, I've addressed these comments in https://github.com/10up/plugin-check/pull/51